### PR TITLE
hack: TLS setup using cfssl

### PR DESCRIPTION
**Pull request by gtank**
this demonstrates basic TLS setup with cfssl. it's much easier than other
available tools.

### CODE REVIEW

File: hack/tls-setup/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+This demonstrates using Cloudflare's [cfssl](https://github.com/cloudflare/cfssl) to easily generate certificates for an etcd cluster.
+
+Defaults generate an ECDSA-384 root and leaf certificates for `localhost`.
+
+**Instructions**
+
+1. Install git, go, and make
```
Comment (yichengq) [inline hack/tls-setup/README.md]: and how to install cfssl

File: hack/tls-setup/Procfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Use goreman to run `go get github.com/mattn/goreman`
+etcd1: bin/etcd -name infra1 -listen-client-urls https://localhost:4001 -advertise-client-urls https://localhost:4001 -listen-peer-urls https://localhost:7001 -initial-advertise-peer-urls https://localhost:7001 -initial-cluster-token etcd-cluster-1 -initial-cluster 'infra1=https://localhost:7001,infra2=https://localhost:7002,infra3=https://localhost:7003' -initial-cluster-state new --cert-file=certs/etcd1.pem --key-file=certs/etcd1-key.pem --peer-cert-file=certs/etcd1.pem --peer-key-file=certs/etcd1-key.pem --peer-client-cert-auth --peer-trusted-ca-file=certs/ca.pem
```
Comment (yichengq) [inline hack/tls-setup/Procfile]: make it runnable in this dir directly as we do in hack/discovery? it is more convenient then. no need to copy files around

File: hack/tls-setup/README.md (hunk 1 above)
Comment (yichengq) [inline hack/tls-setup/README.md]: i see the make does include that, but maybe it is not run correctly. here is what i get:
```
mkdir -p certs
cfssl gencert -initca config/ca-csr.json | cfssljson -bare certs/ca
/bin/sh: cfssl: command not found
/bin/sh: cfssljson: command not found
make: *** [ca] Error 127
```

Comment (gtank) [inline hack/tls-setup/README.md]: Ah, I have $GOPATH/bin in my PATH. I'll add that part.

**Comment by gtank** [comment]
@yichengq I fixed the Procfile to run from `hacks/`. Does the cfssl installation work for you now?

File: hack/tls-setup/Makefile [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.PHONY: cfssl ca req clean
+
+CFSSL	= @env PATH=$(GOPATH)/bin:$(PATH) cfssl
+
+all: cfssl ca req
+
+cfssl:
+	go get -u github.com/cloudflare/cfssl/cmd/cfssl
```
Comment (yichengq) [inline hack/tls-setup/Makefile]: do we need to install `cfssljson` too?

**Comment by yichengq** [comment]
LGTM. It works well for me.

Please ping me next time that you update the commit. github will not notify me if you don't write something.

**Comment by yichengq** [comment]
Could you also add this method into https://github.com/coreos/etcd/blob/master/Documentation/security.md ? So people could notice this easy solution.

**Comment by yichengq** [comment]
I will add the hook in existing docs. merging it.

**Pull request merged**